modules/core/src/smithy4s/http/internals/package.scala

```diff
@@ -24,6 +24,10 @@ package object internals {
   private[http] type HttpCode[A] = A => Option[Int]
   private[http] val httpHints = HintMask(HttpBinding)
 
+  private[internals] type HostPrefixEncode[A] =
```

Comment:
```suggestion
  private[internals] type HostPrefixEncoder[A] =
```

I'd probably add that the type alias `MaybeHostPrefixEncode` is not particularly useful